src/coreclr/jit/hwintrinsiclistxarch.h

```diff
@@ -1374,6 +1374,141 @@ HARDWARE_INTRINSIC(AVX512F,         ShiftLeftMask,
 HARDWARE_INTRINSIC(AVX512F,         ShiftRightMask,                             -1,              2,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid},           HW_Category_IMM,                    HW_Flag_FullRangeIMM|HW_Flag_SpecialCodeGen)
 HARDWARE_INTRINSIC(AVX512F,         XorMask,                                    -1,              2,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid},           HW_Category_Special,                HW_Flag_NoContainment|HW_Flag_Commutative|HW_Flag_ReturnsPerElementMask)
 
+// ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//                 ISA              Function name                               SIMD size       NumArg  EncodesExtraTypeArg                                                                                                       Instructions                                                                                                                  Category                            Flags
+//                                                                                                              {TYP_BYTE,              TYP_UBYTE,              TYP_SHORT,              TYP_USHORT,             TYP_INT,                TYP_UINT,               TYP_LONG,               TYP_ULONG,              TYP_FLOAT,              TYP_DOUBLE}
+// ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//  AVX10V1 Intrinsics
```

Comment:
These should generally be moved up under the `AVX512VBMI_VL` group.

The very bottom of the file is for special intrinsics that are synthesized during lowering/morph. Speaking of which, cases like `NI_AVX512F_KORTEST` or `NI_AVX512F_AddMask` which are synthesized, we should probably rename to something like `NI_EVEX_KORTEST` and `NI_EVEX_AddMask`. It's basically all the entries under the last group labeled `// Special intrinsics that are generated during lowering`, since they are not AVX512 specific and also apply to AVX10v1.

Reply:
For changing the name of the intrinsic, do you mean something like changing -->
```
HARDWARE_INTRINSIC(AVX512F,            KORTEST,                                     0,             -1,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid},           HW_Category_Special,                HW_Flag_NoContainment)
```
to -->
```
HARDWARE_INTRINSIC(EVEX,            KORTEST,                                     0,             -1,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid},           HW_Category_Special,                HW_Flag_NoContainment)
```

If so, do we introduce a new ```InstructionSet_EVEX```? How do we want to handle that?

Reply:
I'm assuming since these(`NI_EVEX_KORTEST` etc) are being generated during lowering, we can just check if Evex encoding is supported and use these?

Reply:
true. but when we declare a hardware intrinsic, it will look for InstructionSet_EVEX and hence I was just clarifying that should we leave it as is or define a new InstructionSet_EVEX?

Reply:
I think I see what you mean here. I disabled ```Avx512*``` and enabled ```Avx10v1``` and set ```PreferredVectorBitWidth=256``` and I see the failures on GNR. These failures are mainly due to introduction of new intrinsics during lowering which you asked to convert to ```EVEX```. I am still thinking about how to fix this but let me know if you have any ideas as to what shall be done.

Reply:
> If so, do we introduce a new InstructionSet_EVEX? How do we want to handle that?

There's a few ways that could be handled and I don't have a particular preference as to which is done, I see some pros/cons for each approach.

You could:
* define some explicit `EVEX` pseudo-instruction set, so defining `InstructionSet_EVEX` just for these helpers.
  * defining a new EVEX instruction set may make some other aspects more confusing; however, we already have some pseudo ISAs like Vector128/256/512 and its worked alright overall
* define these as pseudo helpers under `InstructionSet_Vector128/256/512`
  * this makes it convenient to define but requires redefining the member on each of the vector sizes and handling to pick the right one anywhere they're used
* define new macros that don't require any `InstructionSet_*` to be specified, but still keep the rest of the metadata
  * similar overall considerations to the past two
* remove these entries from the `hwintrinsiclistxarch.h` file entirely and define them under `#ifdef TARGET_XARCH` in `namedintrinsiclist.h` instead
  * this makes it easy to define these special helpers, but makes it hard to use it with flags, instruction lookup, and other helpers that exist for hardware intrinsic APIs; on the other hand, many like KORTEST basically have no relevant metadata anyways (almost all of it is invalid or requires lookup anyways)

Reply:
The main thing is really just making sure that them being under `NI_AVX512F_*` part doesn't cause issues when `Avx10v1` is supported but `Avx512F` is not. There's many ways we could special case them or extract them out to their own concept.

There's some aspects of our overall logic that might get cheaper/simpler if we had some pseudo `InstructionSet_EVEX` that we could say `Avx512F` and `Avx10v1` both depend on and which would in turn possibly simplify the cases we're currently checking `compOpportunisticallyDependsOn(AVX512F || AVX10v1)` which then instead become `compOpportunisticallyDependsOn(EVEX)`, we of course then still need to also check whether `512-bit vectors` are supported to determine if we're using `Avx512F_VL` or `Avx10v1` intrinsics, but that can get consolidated or cached without concerns over the R2R usage tracking, so it seems like maybe the better option?

Interested to hear other input on it and if anyone has ideas on another way we could simplify things while still ensuring the important part of the ISA usage is tracked.

Reply:
Thanks for the detailed explanation @tannergooding. Kushal and I had a quick conversation

I think ideally we just add a new `InstructionSet_EVEX`. This should simplify the whole logic if we just set it while setting either `Avx512F `or `Avx10 `CPUID flags. The one hang-up there is [this](https://github.com/dotnet/runtime/pull/99784#discussion_r1529809346).  Can we change `this`https://github.com/khushal1996/runtime/blob/49707a40d4e9b2a70adb9c01db7fcb664a6901e2/src/coreclr/tools/Common/Compiler/HardwareIntrinsicHelpers.cs#L89 to a long?

Reply:
I believe alternatively we might be able to get away with setting it here : https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/compiler.cpp#L2319 and also when 

```
if (instructionSetFlags.HasInstructionSet(InstructionSet_AVX10v1))
{
    instructionSetFlags.AddInstructionSet(InstructionSet_EVEX);
}
```

I'm going to experiment with this and see if it works

Reply:
@tannergooding We have pushed a commit to handle the EVEX scenarios after testing on GNR. Let me know if this is something we could work with. In the meantime, I am working on resolving the CI failures.

Reply:
Thanks! At a glance it looks reasonable. I should be able to start reviewing this again later tonight or early tomorrow, just catching up on everything I missed last week during Microsoft Build.

Reply:
Sure. Thanks for the update. I will also post a set of scenarios I have tested on GNR.